OpenRA.Mods.Common/Activities/Air/Fly.cs

```diff
@@ -149,16 +149,22 @@ public override bool Tick(Actor self)
 			if (!targetIsHiddenActor && target.Type == TargetType.Actor)
 				lastVisibleTarget = Target.FromTargetPositions(target);
 
-			useLastVisibleTarget = targetIsHiddenActor || !target.IsValidFor(self);
+			// Only update the visibility state when the target is a valid actor, otherwise we keep the visibility state
+			// it had when it was still alive.
```

Comment:
Style nit: the bulk of our inline comments don't end with a full stop, so it looks odd here.

Reply:
Currently it's a mess. There are many comments in the surrounding code which do have a full stop and many which don't&mdash;with no rhyme or reason. I always aim to use proper punctuation in my comments so I prefer to have it here as well.

Reply:
Ok. I'm not going to force you to use incorrect punctuation :smile: